src/mame/drivers/djboy.cpp

```diff
@@ -153,35 +153,27 @@ BS
 
 WRITE8_MEMBER(djboy_state::beast_data_w)
 {
-	m_data_to_beast = data;
+	m_beastlatch->write(space, 0, data);
 	m_z80_to_beast_full = 1;
-	m_beast_int0_l = 0;
-	m_beast->set_input_line(INPUT_LINE_IRQ0, ASSERT_LINE);
-}
-
-READ8_MEMBER(djboy_state::beast_data_r)
-{
-	m_beast_to_z80_full = 0;
-	return m_data_to_z80;
 }
 
 READ8_MEMBER(djboy_state::beast_status_r)
 {
-	return (!m_beast_to_z80_full << 2) | (m_z80_to_beast_full << 3);
+	return (m_slavelatch->pending_r() ? 0x0 : 0x4) | (m_z80_to_beast_full << 3);
```

Comment:
This likely should be `m_beastlatch->pending_r()` rather than the old  `m_z80_to_beast_full` variable, which the code no longer clears and is probably redundant now.